File: src/main/java/org/axostudio/axohologram/animation/DisplayAnimationFrame.java
package org.axostudio.axohologram.animation;

public record DisplayAnimationFrame(
        double offsetX,
        double offsetY,
        double offsetZ,
        float yawOffset,
        float pitchOffset,
        float rollOffset,
        float scaleMultiplier
) {

    public static final DisplayAnimationFrame IDENTITY =
            new DisplayAnimationFrame(0.0D, 0.0D, 0.0D, 0.0F, 0.0F, 0.0F, 1.0F);
}

File: src/main/kotlin/org/axostudio/axohologram/feature/animation/display/DisplayAnimationFrame.kt
package org.axostudio.axohologram.feature.animation.display

data class DisplayAnimationFrame(
    val offsetX: Double = 0.0,
    val offsetY: Double = 0.0,
    val offsetZ: Double = 0.0,
    val yawOffset: Float = 0.0f,
    val pitchOffset: Float = 0.0f,
    val rollOffset: Float = 0.0f,
    val scaleMultiplier: Float = 1.0f
) {
    companion object {
        @JvmField
        val IDENTITY = DisplayAnimationFrame()
    }
}
